### PR TITLE
support for existing password on password change

### PR DESCRIPTION
**Pull request by andrewpai**
Admin UI and API doc changes for https://github.com/FusionAuth/fusionauth-issues/issues/1578

### CODE REVIEW

**Review by spwitt — CHANGES_REQUESTED** [review]
Alphabetize field documentation according to the field name. The current preference is to avoid extra line breaks unless they are required for formatting.

File: site/docs/v1/tech/apis/_generic-update-explanation-fragment.adoc [hunk 1]
```diff
@@ -8,7 +8,9 @@ ifdef::extra_id_verbiage[]
 {extra_id_verbiage}
 endif::[]
 
-You must specify all of the properties of the {capitalized_object_name} when calling this API with the `PUT` HTTP method. When used with `PUT`, this API doesn't merge the existing {capitalized_object_name} and your new data. It replaces the existing {capitalized_object_name} with your new data.
+You must specify all of the properties of the {capitalized_object_name} when calling this API with the `PUT` HTTP method.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/apis/_generic-update-explanation-fragment.adoc]: @andrewpai This is an existing convention. Spencer and I discussed changing this, but Brian and Daniel overruled us. You can view the conversation here: https://inversoft.slack.com/archives/C0407G6V5/p1653503200491019

File: site/docs/v1/tech/apis/_application-request-body.adoc [hunk 2]
```diff
@@ -40,6 +40,19 @@ When this value is not defined, the value defined by [field]#tenant.externalIden
...
++
+:premium_feature: custom forms
+include::../shared/_premium-edition-blurb-api.adoc[]
++
+:premium_feature!:
+
+[field]#application.formConfiguration.requireCurrentPasswordOnPasswordChange# [type]#[Boolean]# [optional]#Optional# [default]#Defaults to `false`# [since]#Available since 1.45.0#::
+When `true`, a user is required to present their current password when changing their password on the self-service account form.
++
+This setting has no effect when a self-service account form has not been set on the Application.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/_application-request-body.adoc:54]: We should also mention this is only enforced with the form has a password field configured.

File: site/docs/v1/tech/apis/users.adoc [hunk 3]
```diff
@@ -211,7 +211,7 @@ include::docs/v1/tech/apis/_user-response-body.adoc[]
 include::docs/v1/tech/apis/_generic-update-explanation-fragment.adoc[]
 :capitalized_object_name!:
 
-If you specify a new password for the User, it will be encrypted and stored. However, if you do not provide a new password, the User's old password will be preserved. This is the only field that is merged during an update using the `PUT` method.
+The one exception to this is with the password field. If you do not specify a password using the `PUT` method, the user's current password will be preserved and will not be deleted. A user's password will only be overwritten if you provide a new password.
```
Comment (robotdan) [inline site/docs/v1/tech/apis/users.adoc:214]: Did you mean to remove the previous copy? "The one exception to this.." seems like it should come after a previous statement which looks to be missing.

File: site/docs/v1/tech/core-concepts/applications.adoc [hunk 4]
```diff
@@ -411,6 +411,9 @@ The form that will be used in the FusionAuth UI for adding and editing user regi
 [field]#User self-service# [optional]#Optional# [since]#Available since 1.26.0#::
 The form that will be used with the hosted login pages for user self-service account management.
 
+[field]#Require current password# [optional]#Optional# [since]#Available since 1.45.0#::
+If set, a user will have to provide their current password when changing their password on the self-service account form. Note that a User self-service form must be selected for this setting to have any effect.
```
Comment (robotdan) [inline site/docs/v1/tech/core-concepts/applications.adoc:415]: We should also mention that this only applies when the form has a password field configured.

**Comment by mooreds** [comment]
@andrewpai ping.

**Comment by mooreds** [comment]
Ooops, I didn't see this open PR, so I did it on Friday ( https://github.com/FusionAuth/fusionauth-site/pull/2190 ) because I kept getting API failed messages :( .

Sorry about the duplicate work.

**Pull request closed without merging**